drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw/host_rpu_umac_if.h

```diff
@@ -748,106 +749,46 @@ struct nrf_wifi_channel {
- *
- * @#define NRF_WIFI_SCAN_FLAG_LOW_PRIORITY: scan request has low priority
- * @#define NRF_WIFI_SCAN_FLAG_FLUSH: flush cache before scanning
- * @#define NRF_WIFI_SCAN_FLAG_AP: force a scan even if the interface is configured
- *	as AP and the beaconing has already been configured. This attribute is
- *	dangerous because will destroy stations performance as a lot of frames
- *	will be lost while scanning off-channel, therefore it must be used only
- *	when really needed
- * @#define NRF_WIFI_SCAN_FLAG_RANDOM_ADDR: use a random MAC address for this scan (or
- *	for scheduled scan: a different one for every scan iteration). When the
- *	flag is set, depending on device capabilities the @MAC and
- *	@MAC_MASK attributes may also be given in which case only
- *	the masked bits will be preserved from the MAC address and the remainder
- *	randomised. If the attributes are not given full randomisation (46 bits,
- *	locally administered 1, multicast 0) is assumed.
- *	This flag must not be requested when the feature isn't supported, check
- *	the nl80211 feature flags for the device.
- * @#define NRF_WIFI_SCAN_FLAG_FILS_MAX_CHANNEL_TIME: fill the dwell time in the FILS
- *	request parameters IE in the probe request
- * @#define NRF_WIFI_SCAN_FLAG_ACCEPT_BCAST_PROBE_RESP: accept broadcast probe responses
- * @#define NRF_WIFI_SCAN_FLAG_OCE_PROBE_REQ_HIGH_TX_RATE: send probe request frames at
- *	rate of at least 5.5M. In case non OCE AP is dicovered in the channel,
- *	only the first probe req in the channel will be sent in high rate.
- * @#define NRF_WIFI_SCAN_FLAG_OCE_PROBE_REQ_DEFERRAL_SUPPRESSION: allow probe request
- *	tx deferral (dot11FILSProbeDelay shall be set to 15ms)
- *	and suppression (if it has received a broadcast Probe Response frame,
- *	Beacon frame or FILS Discovery frame from an AP that the STA considers
- *	a suitable candidate for (re-)association - suitable in terms of
- *	SSID and/or RSSI
- */
-#define NRF_WIFI_SCAN_FLAG_LOW_PRIORITY (1 << 0)
-#define NRF_WIFI_SCAN_FLAG_FLUSH (1 << 1)
-#define NRF_WIFI_SCAN_FLAG_AP (1 << 2)
-#define NRF_WIFI_SCAN_FLAG_RANDOM_ADDR (1 << 3)
-#define NRF_WIFI_SCAN_FLAG_FILS_MAX_CHANNEL_TIME (1 << 4)
-#define NRF_WIFI_SCAN_FLAG_ACCEPT_BCAST_PROBE_RESP (1 << 5)
-#define NRF_WIFI_SCAN_FLAG_OCE_PROBE_REQ_HIGH_TX_RATE (1 << 6)
-#define NRF_WIFI_SCAN_FLAG_OCE_PROBE_REQ_DEFERRAL_SUPPRESSION (1 << 7)
-#define NRF_WIFI_SCAN_FLAG_LOW_SPAN (1 << 8)
-#define NRF_WIFI_SCAN_FLAG_LOW_POWER (1 << 9)
-#define NRF_WIFI_SCAN_FLAG_HIGH_ACCURACY (1 << 10)
-#define NRF_WIFI_SCAN_FLAG_RANDOM_SN (1 << 11)
-#define NRF_WIFI_SCAN_FLAG_MIN_PREQ_CONTENT (1 << 12)
+
+#define NRF_WIFI_SCAN_BAND_2GHZ	(1 << 0)
+#define NRF_WIFI_SCAN_BAND_5GHZ	(1 << 1)
+#define NRF_WIFI_SCAN_BAND_6GHZ	(1 << 2)
 
 /**
  * struct nrf_wifi_scan_params - Scan request parameters.
  *
- * @valid_fields: Indicate which of the following parameters are valid.
+ * @passive_scan: If 1, RPU force passive scan on all channels.
```

Comment:
This is a bit confusing as it's `char` parameter. So do you need to set this to `0x01` or `'1'`? to force passive scan on all channels?
If this is just a boolean variable, a different type would look more logical.

Reply:
I intended to utilize the value 0x1. Modified the comment to incorporate the use of 0x1.

Reply:
Why is this a `char` parameter then? I may not see some implementation detail that would give reasoning for it but doesn't look logical.

Reply:
`short int` is fine if there is no `uint8_t` kind of parameter available in all platforms.